packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -153,6 +156,19 @@ export interface StateMachineProps {
    * @default RemovalPolicy.DESTROY
    */
   readonly removalPolicy?: RemovalPolicy;
+
+  /**
+   * Identifier for KMS key, which can be in the format of key ARN, key ID, alias ARN or alias name
+   * @default - no kmsKeyId is associated
+   */
```

Comment:
Documentation is incorrect. The value is either a `Key` or `Alias` construct, not a string.

Additionally, it should state what the KMS key is used for and that providing a value here is what turns on encryption with a customer managed key. By default, data is encrypted with the AWS owned key.